filters/include/pcl/filters/filter.h

```diff
@@ -218,6 +187,40 @@ namespace pcl
       virtual void
       applyFilter (PointCloud &output) = 0;
 
+    /** \brief implementation of filter method
+      *  Added to ensure no code duplication is present between the with and without
+      * execuotor filter method
+      *
+      * \param[int] filterIndices the callable which calls the filter method
+      * \param[out] output the resultant filtered point cloud
+      */
+      template <typename Callable>
+      void
+      filterImpl (Callable &filterCloud, PointCloud &output)
```

Comment:
Why don't we capture the output in the callable?

Reply:
In indices, we can't capture. So I wanted to ensure the callable had some uniformity.